### ISSUE TITLE
当 dubbo 服务进行分组时， 在服务列表界面点击会报错

### ISSUE THREAD

**Issue by WywTed** [apache/dubbo-admin#61]
javax.servlet.ServletException: Circular view path []: would dispatch back to the current handler URL [/governance/services/b/com.tenxcloud.example.dubbos.CityService/] again. Check your ViewResolver setup! (Hint: This may be the result of an unspecified view, due to default view name generation.)

**Comment by nzomkxia** [comment]
can you provide an URL address

**Comment by soonway** [comment]
根据ServicesController这里定义的URI：/services/$service/providers
在dubbo服务分组时拿到的$service其实是包含了group的URI，但是并没有做包含group的URI映射处理。
比如group是news，则这个URL就像下面的样子
http://xxx.xxx.xxx.xxx/governance/services/news/com.xxx.xxx.facade.service.IconFacede:1.0/providers
我试着在2.5.3的war包里面运行，在服务分组的情况下也是正常的。

**Comment by wangzhangxing** [comment]
me too 啊

**Comment by tanyouxiang** [comment]
我也遇到了,服务分组了以后就不行.

**Comment by yinghuzhu** [comment]
the root cause is that the "ServiceKey" is generated by group + "/"+interface+":"+version  ; the source code located at: com/alibaba/dubbo/common/URL.java:1211 
```
public String getServiceKey() {
        String inf = getServiceInterface();
        if (inf == null) return null;
        StringBuilder buf = new StringBuilder();
        String group = getParameter(Constants.GROUP_KEY);
        if (group != null && group.length() > 0) {
            buf.append(group).append("/");
        }
        buf.append(inf);
        String version = getParameter(Constants.VERSION_KEY);
        if (version != null && version.length() > 0) {
            buf.append(":").append(version);
        }
        return buf.toString();
    }
```
So, I suggest:  "ServiceKey" = group + ":"+interface+":"+version

any idea ?

**Comment by xllove-moon** [comment]
I meet this problem too!

**Comment by cookerzhu** [comment]
The ServiceKey starts with a '/' which will conflict with the url pattern

**Comment by zhuSilence** [comment]
The same as this error. How to solve it?

**Comment by wanxx284105892** [comment]
这个问题现在解决了吗？ @nzomkxia

**Comment by KeRan213539** [comment]
This problem has been fixed by #669.
Please feel free to reopen it if you have any question.
